sdk/src/vote_program.rs

```diff
@@ -81,7 +81,7 @@ pub enum VoteInstruction {
 #[derive(Debug, Default, Serialize, Deserialize, PartialEq, Eq)]
 pub struct VoteState {
     pub votes: VecDeque<Lockout>,
-    pub node_id: Pubkey,
+    pub delegate_id: Pubkey,
     pub staker_id: Pubkey,
```

Comment:
This PR should delete `staker_id`.

Reply:
@garious, thanks for reviewing this pr, I should have put it up as a draft pr because I'm missing tests.

I'm still unclear on how to get rid of staker_id. We need it to credit tokens back to some account. Unless we're assuming it'll go back to the delagted pubkey.

Reply:
Nothing depends on the rewards account right now. You can change it to send the rewards to an arbitrary account. We can let the human staker decide if they want to send those tokens back into their staking account or go buy a lambo.

Reply:
@garious, I think this is resolved per our discord discussion last night. Rewards will get credited to the voting account and get re-staked. The "redemption" account will not be configurable be carried around in the Vote_state. The Vote/Staking account owner can just pull the tokens out into whatever account they want via a regular tx. Sounds ok?

Reply:
Yes, the `from` account used to fund a voting account is not a staking account. That was the super weird thing about the original code.

Reply:
But they won't be able to pull their tokens out of a staking account via a regular transaction.  We'd need to add a `VoteInstruction::Unstake` to pull tokens out.  And there will need to be a cooldown period since the last vote to do that. Otherwise, the staker could do something slashable after pulling their stake since they'd still be in `node_stakes()`.

Reply:
I'll file an issue. How do I block regular transfers on staking accounts? (something in system program I guess? Don't debit from staking accounts)

Also if they pull all their stake, they will no longer be in `node_stakes()` since one of the constraints there is to have non-zero stake.

Reply:
You can't block it. The runtime blocks it. Write that test and see what error gets returned.

No, they'll get boot from the stakers after `stakers_slot_offset`.